classes/s3/helper.php

```diff
@@ -0,0 +1,139 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_lcbackupcoursestep\s3;
+
+use Aws\MockHandler;
+use Aws\Result;
+use stdClass;
+use stored_file;
+use tool_lifecycle\local\manager\settings_manager;
+use tool_lifecycle\settings_type;
+
+/**
+ * Helper class to work with s3 bucket.
+ *
+ * @package     tool_lcbackupcoursestep
+ * @copyright   2024 Catalyst IT
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+
+    /**
+     * Create a new S3 client.
+     *
+     * @param array $settings amazon s3 settings.
+     * @return \Aws\S3\S3Client
+     */
+    public static function create_client($settings) {
```

Comment:
if we expect settings as an array, let's cast array

Reply:
I have fixed this.